waku/node/waku_node.nim

```diff
@@ -223,46 +225,103 @@ proc registerRelayDefaultHandler(node: WakuNode, topic: PubsubTopic) =
     await filterHandler(topic, msg)
     await archiveHandler(topic, msg)
 
-  node.wakuRelay.subscribe(topic, defaultHandler)
+  discard node.wakuRelay.subscribe(topic, defaultHandler)
 
-
-proc subscribe*(node: WakuNode, topic: PubsubTopic) =
+proc subscribe*(node: WakuNode, subscription: SubscriptionEvent) =
+  ## Subscribe to a pubsub topic without a handler.
+  ## Adds default handler to push messages to FILTER & STORE
+  
   if node.wakuRelay.isNil():
     error "Invalid API call to `subscribe`. WakuRelay not mounted."
     return
+  
+  let pubsubTopic =
+    case subscription.kind:
+      of ContentSub:
+        getShard((subscription.contentSub)).valueOr:
+          error "Autosharding error", error=error
+          return
+      of PubsubSub: subscription.pubsubSub
+      else: return
+
+  if node.wakuRelay.isSubscribed(pubsubTopic):
+    error "Invalid API call to `subscribe`. Was already subscribed"
+    return
 
-  debug "subscribe", pubsubTopic= topic
+  debug "subscribe", pubsubTopic=pubsubtopic
 
-  node.topicSubscriptionQueue.emit(SubscriptionEvent(kind: PubsubSub, pubsubSub: topic))
-  node.registerRelayDefaultHandler(topic)
+  node.topicSubscriptionQueue.emit(SubscriptionEvent(kind: PubsubSub, pubsubSub: pubsubtopic))
+  node.registerRelayDefaultHandler(pubsubtopic)
 
-proc subscribe*(node: WakuNode, topic: PubsubTopic, handler: WakuRelayHandler) =
-  ## Subscribes to a PubSub topic. Triggers handler when receiving messages on
-  ## this topic. TopicHandler is a method that takes a topic and some data.
+proc subscribe*(node: WakuNode, subscription: SubscriptionEvent, handler: WakuRelayHandler) =
+  ## Subscribes to a PubSub or Content topic. Triggers handler when receiving messages on
+  ## this topic. WakuRelayHandler is a method that takes a topic and a Waku message.
+  
   if node.wakuRelay.isNil():
     error "Invalid API call to `subscribe`. WakuRelay not mounted."
     return
 
-  debug "subscribe", pubsubTopic= topic
+  let (pubsubTopic, contentTopicOp) =
+    case subscription.kind:
+      of ContentSub:
+        let shard = getShard((subscription.contentSub)).valueOr:
+          error "Autosharding error", error=error
+          return
+
+        (shard, some(subscription.contentSub))
+      of PubsubSub: (subscription.pubsubSub, none(ContentTopic))
+      else: return
+      
+  if contentTopicOp.isSome() and node.contentTopicHandlers.hasKey(contentTopicOp.get()):
+    error "Invalid API call to `subscribe`. Was already subscribed"
+    return
+
+  debug "subscribe", pubsubTopic=pubsubTopic
+
+  node.topicSubscriptionQueue.emit(SubscriptionEvent(kind: PubsubSub, pubsubSub: pubsubTopic))
```

Comment:
80 is not a lot! 100 deal? :rofl:

Reply:
hahaha ! Well, 100 is acceptable IMHO, of course :)
Maybe nowadays doesn't make sense to restrict to 80 chars but limiting code-line length somehow is always beneficial for readability; to compare diffs; and to avoid having to scroll horizontally.

Example of the "80-chars" rule in `nim-libp2p`: https://github.com/status-im/nim-libp2p/blob/74c402ed9db4652a455c00c8d1713b222e3ef3d5/libp2p/transports/wstransport.nim

Reply:
I really wish we had an autoformatter tool!